#! /bin/sh
patch -p1 -f $* < $0
exit $?

Fix a couple of buglets in the I/O channel implementation.

(no changelog yet)

diff -rup glib-2.12.12-old/glib/giowin32.c glib-2.12.12/glib/giowin32.c
--- glib-2.12.12-old/glib/giowin32.c	2007-05-20 19:02:40.000000000 +0200
+++ glib-2.12.12/glib/giowin32.c	2007-05-20 19:02:40.000000000 +0200
@@ -385,7 +385,7 @@
 {
   GIOWin32Channel *channel = parameter;
   guchar *buffer;
-  guint nbytes;
+  gint nbytes;
 
   g_io_channel_ref ((GIOChannel *)channel);
 
@@ -457,7 +457,7 @@
 
       if (nbytes <= 0)
 	break;
-
+ 
       channel->wrp = (channel->wrp + nbytes) % BUFFER_SIZE;
       if (channel->debug)
 	g_print ("read_thread %#x: rdp=%d, wrp=%d, setting data_avail\n",
@@ -496,7 +496,7 @@
 {
   GIOWin32Channel *channel = parameter;
   guchar *buffer;
-  guint nbytes;
+  gint nbytes;
 
   g_io_channel_ref ((GIOChannel *)channel);
 
@@ -1975,7 +1975,7 @@
 
       fd->fd = (gint) win32_channel->data_avail_event;
 
-      if (win32_channel->thread_id == 0 && (condition & G_IO_IN))
+      if (win32_channel->thread_id == 0)
 	{
 	  if (condition & G_IO_IN)
 	    create_thread (win32_channel, condition, read_thread);

